compiler/natives/src/testing/example.go

```diff
@@ -20,23 +19,12 @@ func runExample(eg InternalExample) (ok bool) {
 
 	// Capture stdout.
 	stdout := os.Stdout
-	r, w, err := os.Pipe()
+	w, err := ioutil.TempFile("", "."+eg.Name+".stdout.")
 	if err != nil {
 		fmt.Fprintln(os.Stderr, err)
 		os.Exit(1)
 	}
 	os.Stdout = w
-	outC := make(chan string)
-	go func() {
-		var buf bytes.Buffer
-		_, err := io.Copy(&buf, r)
-		r.Close()
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "testing: copying pipe: %v\n", err)
-			os.Exit(1)
-		}
-		outC <- buf.String()
-	}()
```

Comment:
Just trying to understand this better. Do you know why the original Go code did this here in a goroutine, instead of simply doing it at the end (original line 51, `out := <-outC`)?

Is it so that writes into the write-end of pipe don't block? I don't know how `io.Pipe` works in that sense. Do you need to read from the read-end, otherwise writing-end blocks?

Ok, reading [its documentation](https://godoc.org/io#Pipe) confirms that's the case:

> Pipe creates a synchronous in-memory pipe. It can be used to connect code expecting an io.Reader with code expecting an io.Writer.
>
> Reads and Writes on the pipe are matched one to one except when multiple Reads are needed to consume a single Write. That is, each Write to the PipeWriter blocks until it has satisfied one or more Reads from the PipeReader that fully consume the written data. The data is copied directly from the Write to the corresponding Read (or Reads); there is no internal buffering.